py/server/deephaven/experimental/iceberg.py

```diff
@@ -80,6 +122,43 @@ def j_object(self) -> jpy.JType:
         return self._j_object
 
 
+class IcebergTable(Table):
+    """
+    IcebergTable is a subclass of Table that allows the users to dynamically update the table with new snapshots from
+    the Iceberg catalog.
+    """
+    j_object_type = _JIcebergTable
+
+    def __init__(self, j_table: jpy.JType):
+        super().__init__(j_table)
+
+    def update(self, snapshot_id:Optional[int] = None):
+        """
+        Updates the table with a specific snapshot. If no snapshot is provided, the most recent snapshot is used.
```

Comment:
"update" is very vague.  Is it appending?  Is it replacing?  etc.

Reply:
Updates the table to match the contents of the specified snapshot. This may result in row removes and additions that will be propagated asynchronously via this IcebergTable's UpdateGraph.